src/ONE.Models.CSharp/Common/Configuration/Spreadsheet/WorksheetView.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections.Generic;
+
+namespace ONE.Models.CSharp.Models.Common.Configuration.Spreadsheet
+{
+    public class WorksheetView : IWorksheetView
+    {
+        public string id { get; set; }
+        public string name { get; set; }
+        public bool? isOwner { get; set; }
+        public EnumWorksheet worksheetType { get; set; }
+
+        public List<uint> columnNumbers { get; set; }
+        public Dictionary<uint, ColumnConfig> columns { get; set; }
+        public string createdOn { get; set; }
+        public bool? canEdit { get; set; }
+        public bool? canDelete { get; set; }
+        public List<dynamic> headers { get; set; } // (ColumnHeader | GroupHeader)
```

Comment:
I'm not sure how this is used but using dynamic as a workaround for the lack of discriminated union in C# seems a bit heavy handed since you've no way of knowing what data structure you're going to be given when looking at an instance of this class and makes unit testing, serializing / deserializing a bit harder.

Might be clearer to pull these out into two different lists (columns, groups) that way the consumer doesn't need to guess at what it is dealing with and someone can't jam the wrong thing in. Otherwise, an interface could be used to at least restrict the types to those that are expected in this collection.

Reply:
Moved to Imposed